### PR TITLE
Backport spotless stuff to basepom

### PR DESCRIPTION
**Pull request by Xcelled**
This copies the relevant bits out of our internal pom to use spotless with our OSS projects.

@stevie400 @jaredstehler @suruuK @ggs5427 @PtrTeixeira @tkindy

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2082,18 +2082,47 @@
         <plugin>
           <groupId>com.hubspot.maven.plugins</groupId>
           <artifactId>prettier-maven-plugin</artifactId>
-          <version>0.22</version>
-          <configuration>
+          <version>1.0-deprecate-SNAPSHOT</version>
```
Comment (jaredstehler) [inline pom.xml:2085]: we can't reference SNAPSHOT versions in the basepom; will need to cut a release of the plugin from this branch.

File: pom.xml [hunk 2]
```diff
@@ -2486,6 +2599,26 @@
                 <npmPath>/opt/build-deps/node-${basepom.prettier.node-version}/lib/node_modules/npm/bin/npm-cli.js</npmPath>
               </configuration>
             </plugin>
+            <plugin>
+              <groupId>com.diffplug.spotless</groupId>
+              <artifactId>spotless-maven-plugin</artifactId>
+              <version>2.43.0</version>
```
Comment (jaredstehler) [inline pom.xml:2605]: does this need to be `2.44.0.BETA1`?

Comment (Xcelled) [reply pom.xml:2605]: derp

File: pom.xml (hunk 1 above)
Comment (Xcelled) [reply pom.xml:2085]: We can also just not use this branch

File: pom.xml [hunk 3]
```diff
@@ -2486,6 +2599,26 @@
...
+                    <nodeExecutable>/opt/build-deps/node-${basepom.prettier.node-version}/bin/node</nodeExecutable>
+                    <npmInstallCache>${user.home}/.m2/repository/.spotless-npm-install-cache</npmInstallCache>
+                  </prettier>
+                </java>
+              </configuration>
+              <dependencies>
+                <dependency>
+                  <groupId>com.diffplug.spotless</groupId>
+                  <artifactId>spotless-lib</artifactId>
+                  <version>1.0-hubspot-SNAPSHOT</version>
```
Comment (jaredstehler) [inline pom.xml:2618]: we can drop this dep now right?

**Review by jaredstehler — APPROVED** [review]

**Pull request merged**